src/main/java/codechicken/nei/recipe/GuiRecipe.java

```diff
@@ -735,7 +763,9 @@ private int getRecipesPerPage(HandlerInfo handlerInfo) {
     }
 
     public Point getRecipePosition(int recipe) {
-        return new Point(5, 32 + yShift + ((recipe % getRecipesPerPage()) * handlerInfo.getHeight()));
+        return new Point(
+                5,
+                32 - (limitToOneRecipe ? 25 : 0) + yShift + ((recipe % getRecipesPerPage()) * handlerInfo.getHeight()));
```

Comment:
Have you tested this with with any of the handlers that use the height hack (as defined in `heighthackhandlers.cfg`)? These might expect a certain GUI layout, mainly a fixed 16px offset from `guiTop`. This mostly affects tooltips.

This also an area I've worked on in #395.

Reply:
> This mostly affects tooltips.

OK, this probably doesn't matter for this features as you won't be seing tooltips anyway.